### PR TITLE
Add a Function to Search Log for Errors

### PR DESCRIPTION
**Pull request by willwill96**

### CODE REVIEW

File: vars/getFailureMessage.groovy [hunk 1]
```diff
@@ -0,0 +1,63 @@
+def call(String log) {
+	script {
```
Comment (willwill96) [inline vars/getFailureMessage.groovy]: Looking into formatting issues

**Comment by LinkMJB** [comment]
@willwill96 do you have any examples/results of this in use that we can review?

**Comment by LinkMJB** [comment]
I also think we should add some logic to attempt to identify errors that we don't have a known "reason" for. This is what I use to help me quickly find build failure causes in maven build output:

```
    echo "=== START SUSPECT ERROR/FAILURE LINES ===";
    egrep -n "\[ERROR\]|Failures: [1-9]|Errors: [1-9]" ${TMPFILE};
    echo "=== END SUSPECT ERROR/FAILURE LINES ===";
    echo "=== START JAVA EXCEPTION LINES ===";
    grep -in "java\.*.*exception" ${TMPFILE};
    echo "=== END JAVA EXCEPTION LINES ===";
```

**Comment by willwill96** [comment]
There are some examples but they are in private repos at the moment

**Comment by LinkMJB** [comment]
> Something to think about the future is adding testing for this function. Could use templates of different log files/errors because they may change over time.

Hey @shaundmorris I think we're hoping that https://github.com/connexta/github-utils-shared-library/pull/2/files#diff-7fed725877c430f28b5b61e25e52a360R39 will serve as a catch-all, and hopefully help us to identify other common failure causes that we can write better rationale to after seeing enough of a repeat cause.

**Pull request merged**